CRM/Core/BAO/CustomField.php

```diff
@@ -83,6 +83,28 @@ public static function &dataType() {
     return self::$_dataType;
   }
 
+  /**
+   * Build the map of custom field's data types and there respective Util type
+   *
+   * @return array
+   *   Data data-type => CRM_Utils_Type
+   */
+  public static function dataToType() {
+    return [
+      'String' => CRM_Utils_Type::T_STRING,
+      'Int' => CRM_Utils_Type::T_INT,
+      'Float' => CRM_Utils_Type::T_FLOAT,
+      'Money' => CRM_Utils_Type::T_FLOAT,
+      'Memo' => CRM_Utils_Type::T_TEXT,
+      'Date' => CRM_Utils_Type::T_DATE,
+      'Boolean' => CRM_Utils_Type::T_BOOLEAN,
+      'StateProvince' => CRM_Utils_Type::T_INT,
+      'Country' => CRM_Utils_Type::T_INT,
+      'Link' => CRM_Utils_Type::T_STRING,
+      'ContactReference' => CRM_Utils_Type::T_INT,
+    ];
+  }
+
```

Comment:
@monishdeb there is a function in the api in utils.php which mimics this - with some differences

```
/**
 * Translate the custom field data_type attribute into a std 'type'.
 *
 * @param array $value
 *
 * @return int
 */
function _getStandardTypeFromCustomDataType($value) {
  $dataType = $value['data_type'];
  //CRM-15792 - If date custom field contains timeformat change type to DateTime
  if ($value['data_type'] == 'Date' && isset($value['time_format']) && $value['time_format'] > 0) {
    $dataType = 'DateTime';
  }
  $mapping = array(
    'String' => CRM_Utils_Type::T_STRING,
    'Int' => CRM_Utils_Type::T_INT,
    'Money' => CRM_Utils_Type::T_MONEY,
    'Memo' => CRM_Utils_Type::T_LONGTEXT,
    'Float' => CRM_Utils_Type::T_FLOAT,
    'Date' => CRM_Utils_Type::T_DATE,
    'DateTime' => CRM_Utils_Type::T_DATE + CRM_Utils_Type::T_TIME,
    'Boolean' => CRM_Utils_Type::T_BOOLEAN,
    'StateProvince' => CRM_Utils_Type::T_INT,
    'File' => CRM_Utils_Type::T_STRING,
    'Link' => CRM_Utils_Type::T_STRING,
    'ContactReference' => CRM_Utils_Type::T_INT,
    'Country' => CRM_Utils_Type::T_INT,
  );
  return $mapping[$dataType];
}

```